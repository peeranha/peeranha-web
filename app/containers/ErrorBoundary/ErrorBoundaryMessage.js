import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './messages';

export const ErrorBoundaryMessage = ({ error, errorInfo, reloadApp }) => (
  <div className="alert alert-danger" role="alert">
    <h3 className="alert alert-danger" role="alert">
      <FormattedMessage {...messages.problemWithWebpage} />
    </h3>
    <details>
      {error.toString()}
      <br />
      {errorInfo.componentStack}
    </details>
    <button className="btn btn-link" onClick={reloadApp}>
      <FormattedMessage {...messages.reloadPage} />
    </button>
  </div>
);

ErrorBoundaryMessage.propTypes = {
  error: PropTypes.object,
  errorInfo: PropTypes.object,
  reloadApp: PropTypes.func,
};

export default React.memo(ErrorBoundaryMessage);
