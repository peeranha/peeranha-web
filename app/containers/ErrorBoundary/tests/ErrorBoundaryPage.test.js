import { ErrorBoundaryMessage } from '../ErrorBoundaryMessage';

describe('ErrorBoundaryMessage', () => {
  const props = {
    error: 'error',
    errorInfo: {
      componentStack: 'componentStack',
    },
  };

  it('snapshot test', () => {
    expect(ErrorBoundaryMessage(props)).toMatchSnapshot();
  });
});
