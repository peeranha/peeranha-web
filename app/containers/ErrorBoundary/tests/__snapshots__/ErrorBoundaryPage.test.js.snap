// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundaryMessage snapshot test 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <h3
    className="alert alert-danger"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="A problem occurred with this webpage"
      id="app.containers.ErrorBoundary.problemWithWebpage"
      values={Object {}}
    />
  </h3>
  <details>
    error
    <br />
    componentStack
  </details>
  <button
    className="btn btn-link"
  >
    <FormattedMessage
      defaultMessage="Reload page"
      id="app.containers.ErrorBoundary.reloadPage"
      values={Object {}}
    />
  </button>
</div>
`;
